### PR TITLE
chore: move SDK naming to v1

### PR DESCRIPTION
**Pull request by hperl**
This adapts the https://intranet.ory.sh/docs/dev/api-guidelines/ to Keto to bring the SDK to v1.

BREAKING CHANGES: This changes the naming in the OpenAPI-generated SDKs. The REST and gRPC APIs are unchanged, so old OpenAPI clients still work (and gRPC clients are completely unaffected).

## Related issue(s)

Fixes https://github.com/ory-corp/cloud/issues/3404

## Checklist

- [x] Rename models, params
- [x] Check all comments

### CODE REVIEW

File: internal/namespace/namespacehandler/handler.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func (h *handler) ListNamespaces(ctx context.Context, _ *rts.ListNamespacesReque
 	return &rts.ListNamespacesResponse{Namespaces: apiNamespaces}, nil
 }
 
-// swagger:route GET /namespaces namespaces getNamespaces
+// swagger:route GET /namespaces relationship getNamespaces
```
Comment (aeneasr) [inline internal/namespace/namespacehandler/handler.go:67]: `getNamespaces` is kinda generic and could clash in the future with another system that has a concept of namespaces. How about `getRelationshipNamespaces` or something similar?

File: internal/relationtuple/swagger_definitions.go [hunk 2]
```diff
@@ -70,15 +70,15 @@ type getRelationsParams struct {
...
+// swagger:parameters postHasRelationship createRelationships
 type bodyRelationTuple struct {
 	// in: body
 	Payload ketoapi.RelationQuery
 }
 
 // The basic ACL relation tuple
 //
-// swagger:parameters getCheck deleteRelationTuples
+// swagger:parameters hasRelationship deleteRelationships
```
Comment (aeneasr) [inline internal/relationtuple/swagger_definitions.go:81]: Please follow the naming conventions we established for parameters: https://intranet.ory.sh/docs/dev/api-guidelines#routes

File: internal/relationtuple/swagger_definitions.go [hunk 3]
```diff
@@ -17,13 +17,13 @@ var (
 
 // The patch request payload
 //
-// swagger:parameters patchRelationTuples
+// swagger:parameters patchRelationships
 type patchPayload struct {
 	// in:body
 	Payload []*ketoapi.PatchDelta
```
Comment (aeneasr) [inline internal/relationtuple/swagger_definitions.go:23]: This should also use a "relationship" related naming.

**Review by aeneasr — CHANGES_REQUESTED** [review]

File: internal/check/handler.go [hunk 4]
```diff
@@ -189,7 +189,7 @@ func (h *Handler) postCheckNoStatus(w http.ResponseWriter, r *http.Request, _ ht
 	h.d.Writer().Write(w, r, &RESTResponse{Allowed: allowed})
 }
 
-// swagger:route POST /relation-tuples/check read postCheckMirrorStatus
+// swagger:route POST /relation-tuples/check relationship postHasRelationshipOrError
```
[the remark this answers is not in the document]
Comment (hperl) [reply internal/check/handler.go:192]: @zepatrik do you know?

File: internal/namespace/namespacehandler/handler.go (hunk 1 above)
Comment (hperl) [reply internal/namespace/namespacehandler/handler.go:67]: fixed.

File: internal/relationtuple/swagger_definitions.go (hunk 3 above)
Comment (hperl) [reply internal/relationtuple/swagger_definitions.go:23]: I'm not sure what you mean. It is used in `patchRelationships`.

File: internal/relationtuple/swagger_definitions.go (hunk 2 above)
Comment (hperl) [reply internal/relationtuple/swagger_definitions.go:81]: I'm not sure what you mean here. We are not naming the parameters, we just declare which endpoints they are being used in.

File: internal/relationtuple/swagger_definitions.go (hunk 3 above)
Comment (aeneasr) [reply internal/relationtuple/swagger_definitions.go:23]: `PatchDelta` is a generic name, but refers to a `RelationshipPatchDocument`. The problem is that in the generated OpenAPI  spec, `PatchDelta` appears. I think this should get the appropriate swagger annotation:

```
// swagger:model relationshipPatch
type PatchDelta {
```

File: internal/relationtuple/swagger_definitions.go [hunk 5]
```diff
@@ -17,30 +17,30 @@ var (
...
 }
 
-// swagger:parameters getRelationTuples
+// swagger:parameters getRelationships
 type getRelationsParams struct {
```
Comment (aeneasr) [inline internal/relationtuple/swagger_definitions.go:27]: This struct should be renamed to `getRelationships` and receive a title:

```
// Get Relationships Request Parameters
//
// swagger:parameters getRelationsParams
type getRelationsParams struct {
  // ...
```

File: internal/relationtuple/transact_server.go [hunk 6]
```diff
@@ -197,11 +197,11 @@ func internalTuplesWithAction(deltas []*ketoapi.PatchDelta, action ketoapi.Patch
 	return
 }
 
-// swagger:route PATCH /admin/relation-tuples write patchRelationTuples
+// swagger:route PATCH /admin/relation-tuples relationship patchRelationships
 //
-// # Patch Multiple Relation Tuples
+// # Patch Multiple elationships
```
Comment (aeneasr) [inline internal/relationtuple/transact_server.go:202]:
```suggestion
// # Patch Multiple Relationships
```

**Comment by aeneasr** [comment]
A few more to dos:

* [ ] Please rename `genericError` to `errorGeneric`
* [ ] Please rename `"#/components/schemas/patchDelta"` to `"#/components/schemas/relationshipPatch"`
* [ ] In `createRelationships` rename the request body parameter from `"#/components/schemas/relationQuery"` to `"#/components/schemas/createRelationshipsBody"`
* [ ] `createRelationships` returns a type called ` "#/components/schemas/relationQuery"`. Shouldn't this return a `relationship`?
* [ ] `getRelationshipNamespaces` returns a model called `"#/components/schemas/getRelationshipNamespacesResponse"` but it should be named something along the lines of `relationshipNamespaces`, see examples for naming: https://intranet.ory.sh/docs/dev/api-guidelines/#exemplary-definitions . We generally want to prevent using `response` or `request` in swagger names, because openapi generators add those automatically, and we end up with `getRelationshipNamespacesResponseResponse`

Personal tip: check the resulting openapi.json file for inconsistencies :) And compare it with one from e.g. kratos or hydra.

**Review by aeneasr — CHANGES_REQUESTED** [review]
https://github.com/ory/keto/pull/1138#issuecomment-1323901409

**Comment by aeneasr** [comment]
@hperl can you PTAL into the failing tests and what's required to get this merged? TY

**Review by aeneasr — APPROVED** [review]

**Comment by aeneasr** [comment]
@hperl please add an upgrade guide to: https://www.ory.sh/docs/guides/upgrade/sdk-v1

**Pull request merged**